rskj-core/src/main/java/co/rsk/peg/NonStandardErpRedeemScriptBuilder.java

```diff
@@ -0,0 +1,58 @@
+package co.rsk.peg;
+
+import co.rsk.bitcoinj.core.BtcECKey;
+import co.rsk.bitcoinj.core.Utils;
+import co.rsk.bitcoinj.script.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+
+public class NonStandardErpRedeemScriptBuilder implements ErpRedeemScriptBuilder {
+    private static final Logger logger = LoggerFactory.getLogger(NonStandardErpRedeemScriptBuilder.class);
+
+    @Override
+    public Script createRedeemScriptFromKeys(List<BtcECKey> defaultPublicKeys,
+                                             int defaultThreshold,
+                                             List<BtcECKey> emergencyPublicKeys,
+                                             int emergencyThreshold,
+                                             long csvValue) {
+
+        Script defaultRedeemScript = ScriptBuilder.createRedeemScript(defaultThreshold, defaultPublicKeys);
+        Script emergencyRedeemScript = ScriptBuilder.createRedeemScript(emergencyThreshold, emergencyPublicKeys);
+
+        ErpRedeemScriptBuilderUtils.validateRedeemScriptValues(defaultRedeemScript, emergencyRedeemScript, csvValue);
+
+        byte[] serializedCsvValue = Utils.signedLongToByteArrayLE(csvValue);
+        logger.debug("[createRedeemScriptFromKeys] Creating the redeem script from the scripts");
+        Script redeemScript = createRedeemScriptFromScripts(defaultRedeemScript, emergencyRedeemScript, serializedCsvValue);
+
+        logger.debug("[createRedeemScriptFromKeys] Validating redeem script size");
+        ScriptValidations.validateScriptSize(redeemScript);
+
+        return redeemScript;
+
+    }
+
+    private static Script createRedeemScriptFromScripts(Script defaultRedeemScript,
+                                                        Script emergencyRedeemScript,
+                                                        byte[] serializedCsvValue) {
+
+        ScriptBuilder scriptBuilder = new ScriptBuilder();
+        return scriptBuilder
+            .op(ScriptOpCodes.OP_NOTIF)
+            .addChunks(removeOpCheckMultisig(defaultRedeemScript))
+            .op(ScriptOpCodes.OP_ELSE)
+            .data(serializedCsvValue)
+            .op(ScriptOpCodes.OP_CHECKSEQUENCEVERIFY)
+            .op(ScriptOpCodes.OP_DROP)
+            .addChunks(removeOpCheckMultisig(emergencyRedeemScript))
+            .op(ScriptOpCodes.OP_ENDIF)
+            .op(ScriptOpCodes.OP_CHECKMULTISIG)
+            .build();
+    }
+
+    protected static List<ScriptChunk> removeOpCheckMultisig(Script redeemScript) {
+        return redeemScript.getChunks().subList(0, redeemScript.getChunks().size() - 1);
```

Comment:
Should we also validate that the last chunk is actually the checkmultisig op code before returning?

Reply:
Im not sure. The removal happens in the default and the emergency internal redeem scripts, that are created with the bitcoin's ScriptBuilder. So we can be "sure" they have the OP_CHECKMULTISIG at the end